### ISSUE TITLE
Linux application cross-compile for windows is missing dlls, yet Meson doesn't generate any

### ISSUE THREAD

**Issue by superusercode** [mesonbuild/meson#3418]
Attempting to run my cross-compiled binary gives me this error, where numerous dlls are missing. The entire project is devoid of dlls, so I have no idea what to do. There is no mention of this in the documentation.
```
[rsachan@localhost src]$ wine gradio
0009:err:module:import_dll Library libcairo-gobject-2.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libcairo-2.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgdk-3-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgdk_pixbuf-2.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgio-2.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libglib-2.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgobject-2.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgstpbutils-1.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgstreamer-1.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libgtk-3-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libintl-8.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libjson-glib-1.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libpango-1.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libpangocairo-1.0-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libsoup-2.4-1.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:import_dll Library libsqlite3-0.dll (which is needed by L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe") not found
0009:err:module:attach_dlls Importing dlls for L"Z:\\home\\rsachan\\Documents\\Gradio\\build\\src\\gradio.exe" failed, status c0000135
```
Here is the repo involved and my cross-win.txt files.

https://github.com/haecker-felix/Gradio

**Comment by superusercode** [comment]
Here is my cross_win.txt files, I used mingw32 to build this with all packages installed for it on fedora.

```
[binaries]
c = '/usr/bin/i686-w64-mingw32-gcc'
cpp = '/usr/bin/i686-w64-mingw32-cpp'
ar = '/usr/bin/i686-w64-mingw32-ar'
strip = '/usr/bin/i686-w64-mingw32-strip'
exe_wrapper = 'wine' # A command used to run generated executables.
pkgconfig = '/usr/bin/i686-w64-mingw32-pkg-config'

[properties]
sizeof_int = 4
sizeof_wchar_t = 4
sizeof_void* = 4

alignment_char = 1
alignment_void* = 4
alignment_double = 4

has_function_printf = true

[host_machine]
system = 'windows'
cpu_family = 'x86_64'
cpu = 'i686'
endian = 'little'
```

**Comment by NickeZ** [comment]
On windows I think you need to have all the dlls in PATH or in the directory of the executable.

**Comment by jpakkane** [comment]
> On windows I think you need to have all the dlls in PATH

And on Wine specifically the Wine PATH, not the unix shell's PATH.

**Comment by superusercode** [comment]
@jpakkane The issue is that it doesn't appear that mingw made these dlls, as they aren't in the folder of the executable and outright not in the project directory

**Comment by jpakkane** [comment]
If you are using Fedora dependency packages for Gtk et al, then they are somewhere on the system and you need to tell Wine to add their location to PATH. I don't know why it does not do this by default. For example any program using C++ does not work because `libstdc++` is not in Wine's search path.

**Comment by superusercode** [comment]
My bad, this gave me all the info I needed. https://fedoraproject.org/wiki/Packaging:MinGW?rd=Packaging/MinGW#Libraries_.28DLLs.29

TL;DR dlls are stored in the mingw32/64 sysroot folder.

**Comment by superusercode** [comment]
After compiling this successfully w
